### DISCUSSION TITLE
Early Adopters: Suggested Fix AI

### DISCUSSION THREAD

**Discussion by mitsuhiko** [getsentry/sentry#46802] [Launch Feedback]
Hey there, Sentry users! 🚀

It's Armin, and I'm excited to introduce a brand new experimental feature we've been working on: Suggested Fix! This powerful feature leverages AI to help you diagnose issues and propose solutions, making your debugging experience smoother and more efficient. 🧠💡

To access the "Suggested Fix" feature, just look for the "View Suggestion" on the issue page:

![image](https://user-images.githubusercontent.com/7396/232740421-7cd332a8-7750-459d-b0bb-279b9a3bf951.png)

When you click it, Sentry will send certain parts of the error event to OpenAI for analysis. As we value your privacy and security, we want to assure you that under normal circumstances, we do not send data to third parties undisclosed in our subprocessor list.

To use this feature, your organization must provide consent for data sharing. If you've signed a DPA with us, an organization owner will need to sign the "OpenAI Subprocessor Acknowledgement" in the [Legal & Compliance page of your organization](https://sentry.io/orgredirect/organizations/:orgslug/settings/legal/). Otherwise, individual users will receive a prompt to confirm that data will be sent to a third party. 🔏

Please note that we will never send any event data to OpenAI without explicit instructions from you. Only by clicking the "Suggested Fix" button will selected event data be shared with a third party.

Once you've given it a try, you'll see a suggested fix generated by the AI.

We'd love to hear your thoughts on how you're using the Suggested Fix feature, as well as any feedback on its performance.

[Documentation can be found here](https://docs.sentry.io/product/issues/issue-details/suggested-fix/)

_This text has been generated with GPT-4_

**Comment by 3ventic** [comment]
Is the new Suggested Fix feature available for self-hosted Sentry instances? Thanks!

[AI-generated response]

**Reply by chadwhitacre** [reply]
Almost certainly doable. I've made a [ticket in self-hosted](https://github.com/getsentry/self-hosted/issues/2068) to discuss.

**Comment by everly-blake** [comment]
Do you have more specific examples of what data is sent to OpenAI? Does it include data captured in the error? Source code if we've connected our repositories? We have a BAA with Sentry and need to make sure that Personal Health Information isn't potentially being sent to OpenAI with this feature.

**Reply by danielabrahamsgosh** [reply]
We need this info before we can decide to use it. I can't see much in the docs at the moment...

**Reply by cleptric** [reply]
Take a look at https://docs.sentry.io/product/issues/issue-details/ai-suggested-solution/#data-processing

**Comment by MelodicsVincent** [comment]
Looks promising! As an organisation we are considering opting in to this experimental feature, but first we wouldl ike to know if crash data submitted to OpenAI via Sentry's AI Suggested Solution is used for training OpenAI models? 

I've read Sentry's "OpenAI Subprocessor Acknowledgment" which does not seem to address this particular question? [OpenAI's API data usage policy](https://openai.com/policies/api-data-usage-policies) states that: "data submitted by customers via our API to train or improve our models, unless you explicitly decide to share your data with us for this purpose"
Has Sentry explicitly opted in to share data?

**Reply by cleptric** [reply]
We clarified this here https://docs.sentry.io/product/issues/issue-details/ai-suggested-solution/#data-processing.

> This feature uses the OpenAI API and the event data shared via this feature will not be used to train their AI models.
